compiler-rt/cmake/base-config-ix.cmake

```
@@ -77,9 +77,9 @@ else()
   set(COMPILER_RT_TEST_CXX_COMPILER ${CMAKE_CXX_COMPILER} CACHE PATH "C++ Compiler to use for testing")
 endif()
 
-if("${COMPILER_RT_TEST_COMPILER}" MATCHES "clang[+]*$")
-  set(COMPILER_RT_TEST_COMPILER_ID Clang)
-elseif("${COMPILER_RT_TEST_COMPILER}" MATCHES "clang.*.exe$")
+if ("${COMPILER_RT_TEST_COMPILER}" STREQUAL ${CMAKE_C_COMPILER})
```

```suggestion
if("${COMPILER_RT_TEST_COMPILER}" STREQUAL "${CMAKE_C_COMPILER}")
```

The CMake style is to not have the space after `if` ... unlike the rest of the LLVM code. Additionally, I tend to always quote the parameters to `STREQUAL` as otherwise you may end up with an error if the variable is undefined.